### PR TITLE
Headless spawn

### PR DESCRIPTION
**Pull request by PiZZAD0X**
- changes unit in vehicle spawn to have crew created after vehicle and specific to vehicle slot
- changes to updated framehandler for spawning
- fixes params issue on setAssignedVehicle
- changes compile check for object init value
- adds a manual task override (in case there are no vanilla waypoints)

### CODE REVIEW

File: modules/headless_ai/cfgFunctions/create/fn_createEmptyVehicle.sqf [hunk 1]
```diff
@@ -1,57 +1,59 @@
...
 _vehicle setDamage _damage;
 _vehicle setFuel _fuel;
-_turretMags apply {
-    _x params [["_class",
-"",[""]],["_path",[],[[]]],["_ammo",0,[0]]];
+{
+    _x params [["_class","",[""]],["_path",[],[[]]],["_ammo",0,[0]]];
     _vehicle setMagazineTurretAmmo [_class,_ammo,_path];
-};
+} forEach _turretMags;
```
Comment (EM-Creations) [inline modules/headless_ai/cfgFunctions/create/fn_createEmptyVehicle.sqf:40]: Why did you revert nice things like that?

File: modules/headless_ai/cfgFunctions/create/fn_createEmptyVehicle.sqf [hunk 2]
```diff
@@ -1,57 +1,59 @@
...
+
+if (_storedVars isNotEqualTo []) then {
     //LOG_1("Setting vars: %1",_storedVars);
-    _storedVars apply {
+    {
         _x params ["_varName", "_varValue"];
         _vehicle setvariable [_varName,_varValue];
         //LOG_2("Setting _varName: %1 with: %2",_varName,_varValue);
-    };
+    } forEach _storedVars;
```
Comment (EM-Creations) [inline modules/headless_ai/cfgFunctions/create/fn_createEmptyVehicle.sqf:56]: Regression.

File: modules/headless_ai/cfgFunctions/create/fn_createUnit.sqf [hunk 3]
```diff
@@ -64,67 +58,25 @@ if (_handcuffed) then {
...
+    SETVAR(_unit,customName,_uniqueName);
 };
 
 if (_storedVars isNotEqualTo []) then {
-    {
+    _storedVars apply {
         _x params ["_varName", "_varValue"];
         _unit setvariable [_varName,_varValue];
-    } forEach _storedVars;
+    };
```
Comment (EM-Creations) [inline modules/headless_ai/cfgFunctions/create/fn_createUnit.sqf:68]: Regression.

File: modules/headless_ai/cfgFunctions/set/fn_setStance.sqf [hunk 4]
```diff
@@ -6,14 +6,14 @@ params [
     ["_unitStance", "Auto", [""]]
 ];
 
-LOG_1("setstance: %1",_this);
+//LOG_1("setstance: %1",_this);
 
-private _stance = if ((_groupStance isNotEqualTo "Auto") && {_unitStance isEqualTo "Auto"}) then {
+private _stance = if ((_groupStance != "Auto") && {_unitStance == "Auto"}) then {
```
Comment (EM-Creations) [inline modules/headless_ai/cfgFunctions/set/fn_setStance.sqf:11]: Regression in this file.

Comment (PiZZAD0X) [reply modules/headless_ai/cfgFunctions/set/fn_setStance.sqf:11]: no, case insensitivity check

File: modules/modules.sqf [hunk 5]
```diff
@@ -60,6 +60,13 @@ Additional modules that can be enabled by removing the //
 //#include "start_text\root.sqf"
 //#include "task_control\root.sqf"
 //#include "tracking_markers\root.sqf"
+//#include "start_in_vehicle\root.sqf"
```
[the remark this answers is not in the document]
Comment (PiZZAD0X) [reply modules/modules.sqf:63]: I think it is just a case of a line ending change... it was just a result of github merge

File: modules/headless_ai/cfgFunctions/set/fn_setStance.sqf (hunk 4 above)
Comment (EM-Creations) [reply modules/headless_ai/cfgFunctions/set/fn_setStance.sqf:11]: Ok fair enough, didn't realise this made a case difference.

File: modules/headless_ai/cfgFunctions/create/fn_createEmptyVehicle.sqf (hunk 1 above)
Comment (PiZZAD0X) [reply modules/headless_ai/cfgFunctions/create/fn_createEmptyVehicle.sqf:40]: port of some mmfw functions

File: modules/modules.sqf (hunk 5 above)
[the remark this answers is not in the document]
Comment (EM-Creations) [reply modules/modules.sqf:63]: Yeah that's weird and it looks fine if I view the whole file.

File: modules/headless_ai/cfgFunctions/create/fn_createEmptyVehicle.sqf (hunk 1 above)
Comment (EM-Creations) [reply modules/headless_ai/cfgFunctions/create/fn_createEmptyVehicle.sqf:40]: You know better than me, is there a noticeable performance impact by leaving it as forEach or should we just wait for next update to gradually convert?

Comment (PiZZAD0X) [reply modules/headless_ai/cfgFunctions/create/fn_createEmptyVehicle.sqf]: already changed

**Review by EM-Creations — APPROVED** [review]

**Pull request merged**